pkg/kudoctl/cmd/diagnostics/processing_context.go

```diff
@@ -0,0 +1,57 @@
+package diagnostics
+
+import (
+	"fmt"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// processingContext - shared data for the resource collectors
+// provides property accessors allowing to define a collector before the data it needs is available
+// provides update callback functions. callbacks panic if called on a wrong type of runtime.object
+type processingContext struct {
+	podNames      []string
+	root          string
+	opName        string
+	opVersionName string
+	instanceName  string
+}
+
+func (ctx *processingContext) attachToRoot() string {
+	return ctx.root
+}
+
+func (ctx *processingContext) attachToOperator() string {
+	return fmt.Sprintf("%s/operator_%s", ctx.root, ctx.opName)
+}
+
+func (ctx *processingContext) attachToInstance() string {
+	return fmt.Sprintf("%s/instance_%s", ctx.attachToOperator(), ctx.instanceName)
+}
+
+func (ctx *processingContext) mustSetOperatorNameFromOperatorVersion(o runtime.Object) {
+	ctx.opName = o.(*v1beta1.OperatorVersion).Spec.Operator.Name
+}
+
+func (ctx *processingContext) mustSetOperatorVersionNameFromInstance(o runtime.Object) {
+	ctx.opVersionName = o.(*v1beta1.Instance).Spec.OperatorVersion.Name
+}
+
+func (ctx *processingContext) mustAddPodNames(o runtime.Object) {
```

Comment:
```suggestion
func (ctx *processingContext) addPodNames(o runtime.Object) {
```

Reply:
`Why must`:
this prefix seems more or less standard for unsafe methods that might _panic_. In this case it's more a precaution for the calling code changes or if somebody makes a bad decision to re-use it. On the contrast, plain setter is expected never to panic no matter what we try.
`Why not check`:
unsafe type cast is only "syntactically unsafe" here, because it's just a result of generic resource collection where we don't have _generics_ provided by language. It's an _internal_ function and we fully control it's invocation and know for sure the provided argument's type.

Reply:
Hmmm. It's a weird convention that doesn't really add much to my understanding of the code, but ok. I would still prefer not having that prefix, but it's not a blocker.

I agree with the unchecked cast btw, that's just how it is in Go-World...

Reply:
Could be just in my head.
Immediately comes to mind:
[regexp.Mustcompile](https://golang.org/pkg/regexp/#MustCompile)
Some can be found in Kubernetes:
https://github.com/kubernetes/kubernetes/blob/c7a65ca0c352d58d94327ed80f35757bef3d6792/staging/src/k8s.io/component-base/metrics/registry.go#L115-L121
https://github.com/kubernetes/kubernetes/blob/c981c65c9013306d9a65aedbd30479b932e427b2/staging/src/k8s.io/apimachinery/pkg/api/resource/quantity.go#L124
https://github.com/kubernetes/kubernetes/blob/c981c65c9013306d9a65aedbd30479b932e427b2/staging/src/k8s.io/apimachinery/pkg/util/runtime/runtime.go#L159-L160
So I thought there is a certain pattern.
I am OK to rename, if I have some "final" confirmation